test/statistics/StatisticsTest.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var expect = require('chai').expect;
+var Promise = require('bluebird');
+
+var RC = require('../RC');
+var Client = require('../../').Client;
+var Util = require('../Util');
+var Config = require('../../').Config;
+var BuildInfoLoader = require('../../lib/BuildInfoLoader').BuildInfoLoader;
+
+
+describe('Statistics with default period', function () {
+
+    var cluster;
+    var client;
+    var map;
+    var member;
+
+    before(function () {
+        return RC.createCluster(null, null).then(function (res) {
+            cluster = res;
+        }).then(function () {
+            return RC.startMember(cluster.id);
+        }).then(function (m) {
+            member = m;
+            var cfg = new Config.ClientConfig();
+            var ncc = new Config.NearCacheConfig();
+            ncc.name = 'nearCachedMap*';
+            ncc.invalidateOnChange = false;
+            cfg.nearCacheConfigs['nearCachedMap*'] = ncc;
+            cfg.properties['hazelcast.client.statistics.enabled'] = true;
+            return Client.newHazelcastClient(cfg);
+        }).then(function (cl) {
+            client = cl;
+            map = client.getMap('nearCachedMap1');
+        });
+    });
+
+    after(function () {
+        map.destroy();
+        client.shutdown();
+        RC.shutdownCluster(cluster.id);
+    });
+
+    it('should be enabled via configuration', function () {
+        return Util.promiseWaitMilliseconds(1000).then(function () {
+            return getClientStatisticsFromServer(cluster);
+        }).then(function (stats) {
+            expect(stats).to.not.equal('');
+        });
+    });
+
+    it('should contain statistics content', function () {
+        return Util.promiseWaitMilliseconds(1000).then(function () {
+            return getClientStatisticsFromServer(cluster);
+        }).then(function (stats) {
+            expect(stats).to.not.be.null;
```

Comment:
what about if exact number of stats exist or if there are more or less, can you test for it?

Reply:
It can't be less because `contains` tests are for all stats. However I can write for more stats case.

Reply:
I updated the contains method. It also checks whether a stat exists once or not.